metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -134,28 +134,36 @@ final class BloopServers(
       maybeBloopJvmProperties: Option[List[String]],
       maybeJavaHome: Option[String]
   ): Try[Unit] = Try {
+    if (maybeJavaHome.isDefined || maybeBloopJvmProperties.isDefined) {
+      val jvmPropertiesString =
+        (maybeBloopJvmProperties, maybeJavaHome) match {
+          case (None, Some(javaHome)) =>
+            ujson.write(
+              ujson.Obj(
+                "javaHome" -> javaHome
+              )
+            )
+          case (Some(javaOptions), Some(javaHome)) =>
+            ujson.write(
+              ujson.Obj(
+                "javaHome" -> javaHome,
+                "javaOptions" -> javaOptions.map(ujson.Str(_))
+              )
+            )
+          case (Some(javaOptions), None) =>
+            ujson.write(
+              ujson.Obj(
+                "javaOptions" -> javaOptions.map(ujson.Str(_))
+              )
+            )
```

Comment:
There is a way to simlify duplication inside pattern matching
```scala
      val fields: List[(String, ujson.Value)] =
        List(
          maybeJavaHome.map(v => "javaHome" -> ujson.Str(v)),
          maybeBloopJvmProperties.map(opts =>
            "javaOptions" -> ujson.Arr(opts.map(ujson.Str(_)))
          )
        ).flatten
      val obj = ujson.Obj.from(fields)
      val propertiesString = ujson.write(obj)
```

Reply:
I was thinking about it, but it seemed a bit clearer this way with everything written explicitely.

Reply:
Ok sure, not a big deal.

Reply:
I’m afraid the format ujson might use for producing the json might not be exactly compatible with the form in Bloop docs. I remember that’s manual serialization was used in the code in the first place for producing the json. It’s good to double check the produced result and check also if it can be properly consumed both during reading and by Bloop.

Reply:
Changed!

Reply:
We can't anyway use manual serialization (as that leads to a lot of issues) and since this a valid json it should all work correctly. I double checked it. Also, Bloop actually uses a json library for reading, so this should be ok.

Reply:
Please, let me pull the branch and double check if it can find the expected value.

Reply:
Sorry Tomasz, in local testing of the branch, Metals cannot detect that the value written in the Bloop json 
```
{"javaOptions": [-Xmx1G]}
```
is the same as the one assigned to Bloop Jvm Properties as `-Xmx1G` and shows the popup!
To reproduce, first manually change the file to the content shown above. then go to settings and set bloop jvm properties to `Xmx1G`
The expected behavior is to do nothing. Instead a popup dialogue is shown.

Reply:
Oh, it is written automatically with quotation marks, but when I put the quotes there:

```
{"javaOptions": ["-Xmx1G", "Xms100M”]}
```
it still shows the dialogue even if the requested value is equal to the content of the file.
Not sure why.

Reply:
I think it is unrelated to json. Now I am trying to find the root of the issue, since this was supposed to be solved in the previous PR.

Reply:
Ok then we will merge this one and if you could explain the other problem in an issue or create a PR that would be awesome!

Reply:
I think I found the issue, mentioned it in the other comment.